### ISSUE TITLE
Cannot convert undefined or null to object

### ISSUE THREAD

**Issue by uliporeggia** [blinktrade/BlinkTradeJS#16]
Hello all, 
I am calling `blinktrade.balance()` and this error is thrown:

```
TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at /home/uli/Projetos/blinktrade_tests/node_modules/blinktrade/lib/baseTransport.js:91:18
    at process._tickCallback (internal/process/next_tick.js:109:7)
```

Anyone else having this problem? Is this an error in the SDK or I am just missing something?

**Comment by cesardeazevedo** [comment]
Hi, thanks for report.
I will take a look, do you have a full code snippet?

**Comment by uliporeggia** [comment]
Thanks!
Yes, the code is just hello world:

```        
const BlinkTradeRest = require("blinktrade").BlinkTradeRest;
const blinktrade = new BlinkTradeRest({
    "prod"     : false,
    "key"      : process.env.FOXBIT_API_KEY,
    "secret"   : process.env.FOXBIT_API_SECRET,
    "currency" : "BRL",
});

blinktrade.balance().then((balance) => {
    console.log(balance);
}).catch((err) => {
    console.log(err)
});
```

**Comment by cesardeazevedo** [comment]
I think you forgot the `brokerId: 4`, which is the broker id of FOXBIT, if you not pass the `brokerId` the default will be 5, that is used on our testnet https://testnet.blinktrade.com

Also, make sure that are not using the production environment for FOXBIT, since you're using `prod: false`, you can access it here:
https://testnet.foxbit.exchange

You can get the list of brokers here
https://blinktrade.com/docs/#brokers

Let me know if you got it.

**Comment by uliporeggia** [comment]
That is it, `brokerId: 4` and `prod: true` solve the problem.
Also I will take a look at testnet.
Thanks again!

**Comment by cesardeazevedo** [comment]
Glad that it solved your problem.
But indeed, the error is not intuitive, we're going to improve error messages and more validations in the future.

**Comment by lucasmezencio** [comment]
I'm experiencing exactly the same error.

Here is my snippet:

```javascript
require('dotenv').config();

let btRest = require('blinktrade').BlinkTradeRest;
let rest = new btRest({
  brokerId: 4, // FOXBIT
  prod: true,
  currency: 'BRL',
  key: process.env.FOX_KEY,
  secret: process.env.FOX_SECRET,
});

rest.balance().then(function (balance) {
  console.log('BALANCE >>>', balance);
}).catch(function (error) {
  console.log('ERROR >>>', error);
});
```

With the following error:

```
ERROR >>> TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at /Users/alucard/projects/www/temp/bot/node_modules/blinktrade/lib/baseTransport.js:91:18
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

Can anyone help me? :)

**Comment by cesardeazevedo** [comment]
@lucasmezencio As far i could tell you, this would only happens when your key/secret is from a different brokerId from that you specified on the constructor, other thing that could happen is when you don't have a verified account, which returns just  a empty balance.

The brokerId isn't actually needed on the rest transport, but it was used to calculate the available balance, i have fixed the calculation of available balance at 6fde079f54479297fc4dd30a7e7961bda4624fa0 and shipped at `v0.0.12`
